bus-mapping/src/precompile.rs

```diff
@@ -28,46 +28,48 @@ pub(crate) fn execute_precompiled(
+                    // so we need to override the behavior of invalid input
+                    match PrecompileCalls::from(address.0[19]) {
+                        PrecompileCalls::Blake2F | PrecompileCalls::Ripemd160 => {
                             (vec![], gas, false, false)
-                        } else {
-                            (return_value, gas_cost, false, true)
                         }
-                    }
-                    PrecompileCalls::Modexp => {
-                        let (input_valid, [_, _, modulus_len]) = ModExpAuxData::check_input(input);
-                        if input_valid {
-                            // detect some edge cases like modulus = 0
-                            assert_eq!(modulus_len.as_usize(), return_value.len());
-                            (return_value, gas_cost, false, true) // no oog error
-                        } else {
-                            (vec![], gas, false, false)
+                        PrecompileCalls::Bn128Pairing => {
+                            if input.len() > N_PAIRING_PER_OP * N_BYTES_PER_PAIR {
+                                (vec![], gas, false, false)
+                            } else {
+                                (return_value.to_vec(), gas_cost, false, true)
+                            }
                         }
+                        PrecompileCalls::Modexp => {
+                            let (input_valid, [_, _, modulus_len]) =
+                                ModExpAuxData::check_input(input);
+                            if input_valid {
+                                // detect some edge cases like modulus = 0
+                                assert_eq!(modulus_len.as_usize(), return_value.len());
+                                (return_value.to_vec(), gas_cost, false, true) // no oog error
+                            } else {
+                                (vec![], gas, false, false)
+                            }
+                        }
+                        _ => (return_value.to_vec(), gas_cost, false, true),
                     }
-                    _ => (return_value, gas_cost, false, true),
+                } else {
+                    (return_value.to_vec(), gas_cost, false, true)
                 }
-            } else {
-                (return_value, gas_cost, false, true)
             }
-        }
-        Err(err) => match err {
-            PrecompileError::OutOfGas => (vec![], gas, true, false),
-            _ => {
-                log::warn!("unknown precompile err {err:?}");
-                (vec![], gas, false, false)
-            }
-        },
-    };
+            Err(err) => match err {
+                PrecompileError::OutOfGas => (vec![], gas, true, false),
+                _ => {
+                    log::warn!("unknown precompile err {err:?}");
+                    (vec![], gas, false, false)
+                }
+            },
+        };
```

Comment:
why these changes are needed?

Reply:
the signature changed, Bytes version are different

Reply:
The signature changed, args and ret are Bytes instead of vec